node/silkworm/trie/intermediate_hashes_test.cpp

```diff
@@ -1,780 +0,0 @@
-    for (size_t i{0}; i < n; ++i) {
-        const evmc::address address{int_to_address(i)};
-        const auto hash{keccak256(address)};
-        hashed_accounts.upsert(db::to_slice(hash.bytes), db::to_slice(two_eth.encode_for_storage()));
-        account_change_table.upsert(db::to_slice(block_key), db::to_slice(address));
-    }
-
-    // Delete the second third of the accounts
-    for (size_t i{n}; i < 2 * n; ++i) {
-        const evmc::address address{int_to_address(i)};
-        const auto hash{keccak256(address)};
-        hashed_accounts.erase(db::to_slice(hash.bytes));
-        account_change_table.upsert(db::to_slice(block_key), db::to_slice(address));
-    }
-
-    // Don't touch the last third of genesis accounts
-
-    // And add some new accounts, each holding 1 ETH
-    for (size_t i{3 * n}; i < 4 * n; ++i) {
-        const evmc::address address{int_to_address(i)};
-        const auto hash{keccak256(address)};
-        hashed_accounts.upsert(db::to_slice(hash.bytes), db::to_slice(one_eth.encode_for_storage()));
-        account_change_table.upsert(db::to_slice(block_key), db::to_slice(address));
-    }
-
-    const auto incremental_root{increment_intermediate_hashes(txn, context.dir().etl().path(), /*from=*/0)};
-
-    const std::map<Bytes, Node> incremental_nodes{read_all_nodes(account_trie)};
-
-    // ------------------------------------------------------------------------------
-    // Take B: generate intermediate hashes for the accounts as of Block 1 in one go,
-    // without increment_intermediate_hashes
-    // ------------------------------------------------------------------------------
-    txn.clear_map(db::open_map(txn, db::table::kHashedAccounts));
-    txn.clear_map(db::open_map(txn, db::table::kAccountChangeSet));
-
-    // Accounts [0,n) now hold 2 ETH
-    for (size_t i{0}; i < n; ++i) {
-        const evmc::address address{int_to_address(i)};
-        const auto hash{keccak256(address)};
-        hashed_accounts.upsert(db::to_slice(hash.bytes), db::to_slice(two_eth.encode_for_storage()));
-    }
-
-    // Accounts [n,2n) are deleted
-
-    // Accounts [2n,4n) hold 1 ETH
-    for (size_t i{2 * n}; i < 4 * n; ++i) {
-        const evmc::address address{int_to_address(i)};
-        const auto hash{keccak256(address)};
-        hashed_accounts.upsert(db::to_slice(hash.bytes), db::to_slice(one_eth.encode_for_storage()));
-    }
-
-    const auto fused_root{regenerate_intermediate_hashes(txn, context.dir().etl().path())};
-
-    const std::map<Bytes, Node> fused_nodes{read_all_nodes(account_trie)};
-
-    // ------------------------------------------------------------------------------
-    // A and B should yield the same result
-    // ------------------------------------------------------------------------------
-    CHECK(fused_root == incremental_root);
-    CHECK(fused_nodes == incremental_nodes);
-}
-
-TEST_CASE("Incremental vs regeneration for storage") {
```

Comment:
There were valuable tests in intermediate_hashes_test.cpp. Is it possible to restore them?

Reply:
They need to be mostly re-written cause the logic differs.
For instance old implementation returned every possible node during movement
```
 cursor.next();	
    CHECK(nibbles_to_hex(*cursor.key()) == "10");	
    cursor.next();	
    CHECK(nibbles_to_hex(*cursor.key()) == "10B1");	
    cursor.next();	
    CHECK(nibbles_to_hex(*cursor.key()) == "10B3");	
    cursor.next();	
    CHECK(nibbles_to_hex(*cursor.key()) == "11");	
    cursor.next();	
    CHECK(nibbles_to_hex(*cursor.key()) == "13");	
    cursor.next();	
    CHECK(nibbles_to_hex(*cursor.key()) == "131");	
    cursor.next();	
    CHECK(nibbles_to_hex(*cursor.key()) == "132");	
    cursor.next();	
    CHECK(nibbles_to_hex(*cursor.key()) == "133");
```
while actual implementation does not

Will close this PR until done

Reply:
I'm not that worried about lower-level test cases like "Cursor traversal 1", but higher-level cases like "Incremental vs regeneration" and "Incremental vs regeneration for storage" are probably worth preserving.

Reply:
It will take long and and a snsible rewrite to be able to expose the proper methods to the tests and to unbind from node_settings, logging features, and remove check of expected root retrieved from canonical block (which tests do not save). Doing it but don't expect results quickly

Reply:
> It will take long and and a sensible rewrite to be able to expose the proper methods to the tests and to unbind from node_settings, logging features, and remove check of expected root retrieved from canonical block (which tests do not save). Doing it but don't expect results quickly

OK, so do you want to restore "Incremental vs regeneration" and other test cases in this PR or in a follow-up one?

Reply:
I already started refactoring. Really wish this PR goes through without need for follow-ups